homeassistant/components/nanoleaf/light.py

```diff
@@ -157,6 +158,15 @@ def max_mireds(self):
         """Return the warmest color_temp that this light supports."""
         return 833
 
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return self._unique_id
+
+    def __hash__(self):
+        """Return a hash of self."""
+        return hash(self.unique_id)
```

Comment:
Why is this added?

Reply:
To be honest, I just copied from the Sonos media player component which I've studied as an exemplar entity. The comparison function seems to have been added in https://github.com/home-assistant/core/pull/21411/files#diff-0f19863b26c5f41a3317b283bd852c6c9a2526ada54a0a8ca94eaff230d65aacR373

If it isn't strictly necessary I'll happily remove it.